### PR TITLE
build04: add hercules-ci.nix

### PR DESCRIPTION
**Pull request by Mic92**

### CODE REVIEW

File: ci.nix [hunk 1]
```diff
@@ -13,6 +13,8 @@ pkgs.nix-community-infra // rec {
   build01-system = build01.system;
   build02 = importNixOS ./build02/configuration.nix "x86_64-linux";
   build02-system = build02.system;
-  build03 = importNixOS ./build02/configuration.nix "x86_64-linux";
+  build03 = importNixOS ./build03/configuration.nix "x86_64-linux";
   build03-system = build03.system;
+  build04 = importNixOS ./build04/configuration.nix "aarch64-linux";
```
Comment (roberth) [inline ci.nix:18]: Now that there's a second node, you need to configure the binary cache in binary-caches.json, so that drvs can be shared between nodes.

Comment (Mic92) [reply ci.nix:18]: Yeah. We noticed this the hard way.

Comment (roberth) [reply ci.nix:18]: I'd like to make it more flexible, but right now the focus is on more high level aspects like a new nix file (including flakes support) and more "typical" CI/CD features.

Comment (Mic92) [reply ci.nix:18]: It's got green https://hercules-ci.com/github/nix-community/infra/jobs/13 eventually.

File: ci.nix [hunk 2]
```diff
@@ -13,6 +13,8 @@ pkgs.nix-community-infra // rec {
...
+  build04-system = build04.system;
 }
```
Comment (roberth) [inline ci.nix:20]: Actually it'd be ok for this file to be `default.nix`. 

> At the top-level of your repository, the agent will search for a Nix file to evaluate in the following order: nix/ci.nix, ci.nix or default.nix

 -- https://docs.hercules-ci.com/hercules-ci-agent/evaluation/
 
 It also documents the traversal of attributes.
 
 Next release series, 0.9.x, is updating this to include flake.nix support among other things.

Comment (roberth) [reply ci.nix:20]: That said, renaming this to `ci.nix` is of course fine if CI is the whole purpose of this file.

Comment (Mic92) [reply ci.nix:20]: Yes, making the intent of this file clear, was my main motivation.

**Pull request merged**